server/src/routers/public/profiles.ts

```diff
@@ -31,57 +31,89 @@ function createQueryWithStatusFilter(statuses, tableName: string) {
     tableName + ".pbo_category_id",
     "category.pbo_category_id"
   );
+
   return query;
 }
 
-function addCountChildrenInnerQuery(query, statuses) {
-  const innerQuery = createQueryWithStatusFilter(statuses, "innerProfile")
+function createChildrenCountQuery(statuses) {
+  return createProfileQueryWithStatusFilter(statuses, "innerProfile")
     .select("innerProfile.id", db.raw("COUNT(child.id) AS childrenCount"))
-    .leftJoin("profiles AS child", "innerProfile.id", "child.parent");
-  innerQuery.groupBy("innerProfile.id").as("counts");
+    .leftJoin("public.profiles AS child", "innerProfile.id", "child.parent")
+    .groupBy("innerProfile.id");
+}
 
-  query.join(innerQuery, "profile.id", "counts.id");
-  return query;
+function addChildrenCountInnerQuery(query, statuses) {
+  const innerQuery = createChildrenCountQuery(statuses).as("counts");
+  return query.join(innerQuery, "profile.id", "counts.id");
+}
+
+function createProfileIdsWithoutPaymentsQuery() {
+  return db("public.profiles AS profiles")
+    .select("profiles.id")
+    .leftJoin(
+      "public.payments AS payments",
+      "profiles.id",
+      "payments.profile_id"
+    )
+    .whereNull("payments.profile_id");
+}
+
+async function setHasPaymentsFlag(profiles: ProfileRecord[]) {
```

Comment:
purpose of this is to fix existing data?

Reply:
No, it just sets the field in the Profile models that were previously loaded from DB. The query that determines if profiles have payments is executed separately. I could not find a more efficient way than to do it in two queries. (It could be done with a messy join I guess, but I doubt it would be more efficient.)

So the method is called `set` because it encapsulates logic (used in multiple endpoints) that performs the necessary DB query and writes the result to the provided Profile models.

Reply:
I guess this might be confusing so I will add a comment to the code with the explanation above.

Reply:
I just realized it is not obvious that `hasPayments` is a computed property, not a column in DB.

Reply:
I added comment and renamed the function to `computeHasPaymentsFlag`.